app/views/questionnaire.py

```diff
@@ -182,483 +180,103 @@ def get_hub_section(schema, questionnaire_store, section_id):
-    questionnaire_store_updater.update_relationship_answer(
-        form.data, current_location.from_list_item_id, current_location.to_list_item_id
-    )
-
-    # The location needs to be removed as we may have previously completed this location
-    if 'action[save_sign_out]' in request.form:
-        questionnaire_store_updater.remove_completed_location()
-        questionnaire_store_updater.save()
-        return redirect(url_for('session.get_sign_out'))
-
-    next_location_url = relationship_router.get_next_location_url(current_location)
-    if next_location_url:
-        questionnaire_store_updater.save()
-        return redirect(next_location_url)
-
-    questionnaire_store_updater.add_completed_location()
-    _update_section_completeness(questionnaire_store_updater, routing_path)
-    questionnaire_store_updater.save()
-
-    router = Router(
-        schema, questionnaire_store.progress_store, questionnaire_store.list_store
-    )
-    next_location_url = router.get_next_location_url(parent_location, routing_path)
-    return redirect(next_location_url)
-
-
-def validate_location(schema, routing_path, list_store, current_location):
-    router = Router(schema, get_progress_store(current_user), list_store)
-
-    if not schema.get_block(current_location.block_id):
-        if schema.is_hub_enabled():
-            return url_for('.get_hub')
-
-        redirect_location = router.get_first_incomplete_location_in_survey()
-        return redirect_location.url()
 
-    is_list_collector_child = schema.is_block_list_collector_child(
-        current_location.block_id
-    )
-
-    block = schema.get_block(current_location.block_id)
-    if is_list_collector_child:
-        parent_block_id = schema.get_list_collector_for_block_id(
-            current_location.block_id
-        )['id']
-        routing_location = Location(block_id=parent_block_id)
-    elif block['type'] == 'RelationshipCollector':
-        routing_location = Location(block_id=current_location.block_id)
-    else:
-        routing_location = current_location
-
-    if not router.can_access_location(routing_location, routing_path):
-        redirect_location = router.get_first_incomplete_location_in_survey()
-        return redirect_location.url()
-
-    if is_list_collector_child:
-        return _validate_list_collector_child_location(
-            schema, routing_path, list_store, current_location
-        )
-
-
-def _validate_list_collector_child_location(
-    schema, routing_path, list_store, current_location
-):
-    list_item_id = current_location.list_item_id
-    list_name = current_location.list_name
-    block = schema.get_block(current_location.block_id)
-    parent_block = schema.get_list_collector_for_block_id(current_location.block_id)
-
-    if list_name != parent_block['for_list']:
-        logger.info(
-            f'Mismatched list_name: {list_name} and block_id: {current_location.block_id}'
+    try:
+        block_handler = BlockFactory.get_handler(
```

Comment:
```suggestion
        block_handler = get_handler(
```

Reply:
Please don't comment on every line that would need to be changed, it makes for a very noisy pull request!